lib/Controller/UsersController.php

```diff
@@ -469,6 +500,213 @@ public function create($username, $password, array $groups= [], $email='') {
 		);
 	}
 
+	/**
+	 * Set password for user using link
+	 *
+	 * @PublicPage
+	 * @NoCSRFRequired
+	 * @NoAdminRequired
+	 * @NoSubadminRequired
+	 *
+	 * @param $token
+	 * @param $userId
+	 * @return TemplateResponse
+	 */
+	public function setPasswordForm($token, $userId) {
+		try {
+			$this->checkPasswordSetToken($token, $userId);
+		} catch (UserTokenException $e) {
+			if ($e->getTokeFailureType() === UserTokenException::TOKEN_EXPIRED) {
+				return new TemplateResponse(
+					'user_management', 'new_user/resendtokenbymail', [
+						'link' => $this->urlGenerator->linkToRouteAbsolute('user_management.Users.resendToken', ['userId' => $userId]),
+						'userId' => $userId
+					], 'guest'
+				);
+			}
+			$this->log->error('Caught exception ' . $e->getMessage());
```

Comment:
I think the logger has a `logException` method or something similar. Do we want to log the whole stacktrace or just log the message? If this is something that shouldn't happen and it won't be common, the more information we have, the better.
Note that, if you're catching an specific exception, it seems wrong that you don't know what is happening (which is what this code suggests)

Reply:
Instead of using the $e->getMesage() , I have used `logException` in methods `setPassword` and `setPasswordForm`